tests/integration/sharding_tests/helpers.py

```diff
@@ -1,36 +1,53 @@
 #!/usr/bin/env python3
 # Copyright 2023 Canonical Ltd.
 # See LICENSE file for licensing details.
-from typing import List, Optional
+from typing import List, Optional, Tuple
 from urllib.parse import quote_plus
 
 from pymongo import MongoClient
 from pytest_operator.plugin import OpsTest
 
 from ..helpers import get_password
+from ..relation_tests.new_relations.helpers import (
+    get_application_relation_data,
+    get_secret_data,
+)
 
 MONGOS_PORT = 27018
 MONGOD_PORT = 27017
 
 
-async def generate_mongodb_client(ops_test: OpsTest, app_name: str, mongos: bool):
+async def generate_mongodb_client(
+    ops_test: OpsTest, app_name: str, mongos: bool, username: str = None, password: str = None
+):
     """Returns a MongoDB client for mongos/mongod."""
     hosts = [unit.public_address for unit in ops_test.model.applications[app_name].units]
-    password = await get_password(ops_test, app_name=app_name)
+    password = password or await get_password(ops_test, app_name=app_name)
+    username = username or "operator"
```

Comment:
just curious why not set default `username` in `generate_mongodb_client`

Reply:
great point! thank you!